File: containsDuplicate/contains_duplicate.go
func containsDuplicate(nums []int) bool {
    d := make(map[int]int)
    for _, num := range nums {
        _, ok := d[num]
        if ok {
            d[num] += 1
        } else {
            d[num] = 1
        }
        if d[num] > 1 {
            return true
        }
    }
    return false
    
}

File: containsDuplicate/contains_duplicate.rs
use std::collections::HashMap;

impl Solution {
    pub fn contains_duplicate(nums: Vec<i32>) -> bool {
        let mut counter: HashMap<String, i32> = HashMap::new();
        for num in nums {
            let key = num.to_string();
            let count = counter.entry(key.clone()).or_insert(0);
            *count += 1;
            if (*count > 1) {
                return true;
            }
        }
        return false;
        
    }
}